packages/core/client.js

```diff
@@ -273,14 +273,18 @@ class Client {
     event._user = assign({}, event._user, this._user)
     event.breadcrumbs = this._breadcrumbs.slice()
 
-    if (this._session) {
-      this._session._track(event)
-      event._session = this._session
+    const trackSession = event => {
+      if (this._session) {
+        this._session._track(event)
+        event._session = this._session
+      }
     }
 
     // exit early if events should not be sent on the current releaseStage
     if (this._config.enabledReleaseStages !== null && !includes(this._config.enabledReleaseStages, this._config.releaseStage)) {
       this._logger.warn('Event not sent due to releaseStage/enabledReleaseStages configuration')
+      trackSession(event)
```

Comment:
I think this is unnecessary since the notifier will never send events nor sessions with this configuration.